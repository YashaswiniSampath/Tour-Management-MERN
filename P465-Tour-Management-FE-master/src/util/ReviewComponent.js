import React, { useState } from 'react';

const ReviewComponent = ({ location, onClose, onSubmit  }) => {
    const [reviewText, setReviewText] = useState('');
    const [rating, setRating] = useState(0);

    const handleReviewSubmit = () => {
        console.log("Here is my location", location)
        onSubmit(location._id, reviewText, rating);
        onClose();
    }

    return (
        <div className="modal-backdrop">
            <div className="review-modal">
                <h2>Leave a Review for {location.name}</h2>
                <div className="review-input">
                <textarea
                    value={reviewText}
                    onChange={(e) => setReviewText(e.target.value)}
                    placeholder="Write your review here..."
                />
                <input
                    type="number"
                    value={rating}
                    onChange={(e) => setRating(e.target.value)}
                    min="1"
                    max="5"
                    placeholder="Rating (1-5)"
                />
                <button onClick={handleReviewSubmit} style={{ backgroundColor: '#2484BF', color: 'white'}}>Submit Review</button>
                <button onClick={onClose} style={{ backgroundColor: '#2484BF', color: 'white'}}>Close</button>
                </div>
            </div>
        </div>
    );
};

export default ReviewComponent;
