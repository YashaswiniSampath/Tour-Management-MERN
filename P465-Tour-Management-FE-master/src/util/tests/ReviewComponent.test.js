import renderer from 'react-test-renderer';
import ReviewComponent from '../ReviewComponent';

it('renders page', () => {
    const component = renderer.create(
        <ReviewComponent location={{name: "Test name"}}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})