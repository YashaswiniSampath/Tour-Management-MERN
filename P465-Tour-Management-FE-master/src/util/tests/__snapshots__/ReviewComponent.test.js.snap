// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page 1`] = `
<div
  className="modal-backdrop"
>
  <div
    className="review-modal"
  >
    <h2>
      Leave a Review for 
      Test name
    </h2>
    <div
      className="review-input"
    >
      <textarea
        onChange={[Function]}
        placeholder="Write your review here..."
        value=""
      />
      <input
        max="5"
        min="1"
        onChange={[Function]}
        placeholder="Rating (1-5)"
        type="number"
        value={0}
      />
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2484BF",
            "color": "white",
          }
        }
      >
        Submit Review
      </button>
      <button
        style={
          Object {
            "backgroundColor": "#2484BF",
            "color": "white",
          }
        }
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
